File: db4o.net/Db4oUnit.Extensions/Db4oUnit.Extensions/Db4oAssert.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using System;
using Db4oUnit;
using Db4oUnit.Extensions;
using Db4objects.Db4o.Ext;

namespace Db4oUnit.Extensions
{
	public class Db4oAssert
	{
		public static void PersistedCount(int expected, Type extent)
		{
			Assert.AreEqual(expected, Db().Query(extent).Count);
		}

		private static IExtObjectContainer Db()
		{
			return Fixture().Db();
		}

		private static IDb4oFixture Fixture()
		{
			return Db4oFixtureVariable.Fixture();
		}
	}
}

File: db4ounit.extensions/src/db4ounit/extensions/Db4oAssert.java
/* Copyright (C) 2008  Versant Inc.  http://www.db4o.com */

package db4ounit.extensions;

import com.db4o.ext.*;

import db4ounit.*;


public class Db4oAssert {

    public static void persistedCount(int expected, Class extent) {
        Assert.areEqual(expected, db().query(extent).size());
    }
    
    private static ExtObjectContainer db(){
        return fixture().db();
    }

    private static Db4oFixture fixture() {
        return Db4oFixtureVariable.fixture();
    }
    
    
    
    

}
